contracts/KetherNFTLoaner.sol

```diff
@@ -0,0 +1,258 @@
+ */
+contract KetherNFTLoaner is Ownable {
+  using SafeMath for uint256;
+
+  uint256 private constant _1ETH = 1 ether;
+  uint256 public loanServiceCharge = _1ETH.div(100).mul(5);
+  uint256 public loanChargePerDay = _1ETH.div(1000);
+  uint8 public maxLoanDurationDays = 30;
+  uint8 public loanPercentageCharge = 10;
+  IKetherNFT private _ketherNft;
+
+  struct PlotOwner {
+    address owner;
+    uint256 overrideLoanChargePerDay;
+    uint8 overrideMaxLoanDurationDays;
+  }
+
+  struct PlotLoan {
+    address loaner;
+    uint256 start;
+    uint256 end;
+  }
+
+  struct PublishParams {
+    string link;
+    string image;
+    string title;
+    bool NSFW;
+  }
+
+  mapping(uint256 => PlotOwner) public owners;
+  mapping(uint256 => PlotLoan) public loans;
+
+  event AddPlot(
+    uint256 indexed idx,
+    address owner,
+    uint256 overridePerDayCharge,
+    uint256 overrideMaxLoanDays
+  );
+  event UpdatePlot(
+    uint256 indexed idx,
+    uint256 overridePerDayCharge,
+    uint256 overrideMaxLoanDays
+  );
+  event RemovePlot(uint256 indexed idx);
+  event LoanPlot(uint256 indexed idx, address loaner);
+  event Transfer(address to, uint256 idx);
+
+  constructor(address _ketherNFTAddress) {
+    _ketherNft = IKetherNFT(_ketherNFTAddress);
+  }
+
+  function addPlot(
+    uint256 _idx,
+    uint256 _overridePerDayCharge,
+    uint8 _overrideMaxDays
+  ) external payable {
+    require(
+      msg.sender == _ketherNft.ownerOf(_idx),
+      'You need to be the owner of the plot to loan it out.'
+    );
+    require(
+      msg.value >= loanServiceCharge,
+      'You must send the appropriate service charge to support loaning your plot.'
+    );
+    payable(owner()).call{ value: msg.value }('');
+    _ketherNft.transferFrom(msg.sender, address(this), _idx);
+    owners[_idx] = PlotOwner({
+      owner: msg.sender,
+      overrideLoanChargePerDay: _overridePerDayCharge,
+      overrideMaxLoanDurationDays: _overrideMaxDays
+    });
+    emit AddPlot(_idx, msg.sender, _overridePerDayCharge, _overrideMaxDays);
+  }
+
+  function updatePlot(
+    uint256 _idx,
+    uint256 _overridePerDayCharge,
+    uint8 _overrideMaxDays
+  ) external {
+    PlotOwner storage _owner = owners[_idx];
+    require(
+      msg.sender == _owner.owner,
+      'You must be the plot owner to update information about it.'
+    );
+    _owner.overrideLoanChargePerDay = _overridePerDayCharge;
+    _owner.overrideMaxLoanDurationDays = _overrideMaxDays;
+    emit UpdatePlot(_idx, _overridePerDayCharge, _overrideMaxDays);
+  }
+
+  function removePlot(uint256 _idx) external {
+    address _owner = owners[_idx].owner;
+    require(
+      msg.sender == _owner,
+      'You must be the original owner of the plot to remove it from the loan contract.'
+    );
+    require(
+      !hasActiveLoan(_idx),
+      'There is currently an active loan on your plot that must expire before you can remove.'
```

Comment:
Hm, maybe you should be able to recover your plot if you pay back the original amount+fee?

Reply:
Yeah I like that. I'll add logic to support the owner paying the loaner back a prorated fee if he/she needs to remove their plot from the contract before a loan term is complete.

Reply:
IMO it should be at minimum the full fee rather than prorated. You're breaking an agreement, you shouldn't get any profit from that. The loaner has some expectation of the duration of the term, let's say you line up a whole media campaign with it and the agreement is broken.

Reply:
Understood, this makes sense, I agree and will proceed with ensuring the owner pays back the full amount.

Reply:
Resolved in https://github.com/moontography/contracts/pull/2/files#diff-9ff20691ee2cce9f2a49e4879db943f722ed5ca80e4cbdcf5bc87857fb03cb1eR128